doc/tools/stlink-v21.rst

```diff
@@ -0,0 +1,129 @@
+:orphan:
+
+.. _stlink-v21:
+
+ST-LINK/V2-1
+############
+
+Overview
+********
+
+ST-LINK/V2-1 is a serial and debug adapter built into all Nucleo and Discovery
+boards. It provides a bridge between your computer (or other USB host) and the
+embedded target processor, which can be used for debugging, flash programming,
+and serial communication, all over a simple USB cable.
+
+SEGGER offers a firmware upgrading the ST-LINK/V2-1 on board on the Nucleo and
+Discovery boards. This firmware makes the ST-LINK/V2-1 compatible with J-Link
+OB, allowing users to take advantage of most J-Link features like the ultra fast
+flash download and debugging speed or the free-to-use GDBServer.
+
+More informations about upgrading ST-LINK/V2-1 to JLink or restore ST-Link/V2-1
+fireware please visit: `SEGGER`_
+
+Zephyr supports the following debug tools through ST-LINK/V2-1:
+
+* :ref:`st_ST-LINK/V2-1_stlink`
+* :ref:`st_ST-LINK/V2-1_jlink`
+
+.. _st_ST-LINK/V2-1_stlink:
+
+ST-LINK/V2-1
+************
+
+Flashing
+========
+
+The default debug tool is ST-Link, so no additional flashing or debugging
+command parameters are required:
+
+  .. zephyr-app-commands::
+     :zephyr-app: samples/hello_world
+     :goals: flash
+
+Debugging
+=========
+
+  .. zephyr-app-commands::
+     :zephyr-app: samples/hello_world
+     :goals: debug
+
+
+.. _st_ST-LINK/V2-1_jlink:
+
+Segger J-Link
+*************
+
+Host Tools and Firmware
+=======================
+
+Download and install the `Segger J-Link Software and Documentation Pack`_ to
+get the J-Link GDB server for your host computer.
+
+Flashing
+========
+
+Use the CMake ``flash`` target with the argument ``STLINK_FW=jlink`` to
+build your Zephyr application.
+
+  .. zephyr-app-commands::
+     :zephyr-app: samples/hello_world
+     :gen-args: -DSTLINK_FW=jlink
+     :goals: flash
+
+Debugging
+=========
+
+Use the CMake ``debug`` target with the argument ``STLINK_FW=jlink`` to build
+your Zephyr application, invoke the J-Link GDB server, attach a GDB client, and
+program your Zephyr application to flash. It will leave you at a GDB prompt.
+Some boards set the argument by default, so it is not always necessary to set
+explicitly.
```

Comment:
Does this last sentence apply, since you state in line 37 that ST-Link is the default tool?

Reply:
I agree this line should be removed

Reply:
It's not right because I just copy it from opensda.rst and I havn't notice that. I will fix it. Thanks